plugins/InspectionGadgets/testsrc/com/siyeh/ig/controlflow/NegatedConditionalInspectionTest.java

```diff
@@ -0,0 +1,50 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.siyeh.ig.controlflow;
+
+import com.intellij.codeInspection.InspectionProfileEntry;
+import com.siyeh.ig.LightInspectionTestCase;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * @author Roland Illig
+ */
+public class NegatedConditionalInspectionTest extends LightInspectionTestCase {
+
+  public void testNegatedConditionalNullAndZeroAllowed() {
+    doTest();
+  }
+
+  public void testNegatedConditionalNullAndZeroDisallowed() {
+    doTest();
+  }
+
+  public void testNegatedConditionalNullAllowed() {
+    doTest();
```

Comment:
What you can do is call `myFixture.enableInspections(inspection);` before `doTest()` instead. Then you don't need `getInspection()`, but unfortunately it is `abstract` so a dummy implementation is needed. Do you want to make this change?

Reply:
Thanks for the hint. The test code looks much better now.

I wonder why `enableInspections` has a comment "Should be called BEFORE setUp" when it works well to add an inspection afterwards and it even overrides the previous dummy inspection from `getInspection`.

Reply:
That's weird, because it is actually not possible to call the method before setUp() is called. I'll try to get this fixed.

Also I though of a cleaner way to handle configuration of the inspection under test. Check the manual merge when it becomes available on GitHub.

Reply:
👍 I don't think there's any way to make the test even clearer now.